src/openmrs-backend-dependencies.ts

```diff
@@ -0,0 +1,100 @@
+import { openmrsFetch } from "@openmrs/esm-api";
+import * as semver from "semver";
+import { difference, isEmpty } from "lodash-es";
+
+var installedBackendModules: any[] = [];
+
+const originalOnload = System.constructor.prototype.onload;
+
+System.constructor.prototype.onload = function(err, id, deps) {
+  const moduleName = id.substring(id.lastIndexOf("/") + 1, id.indexOf(".js"));
+  if (!err) {
+    System.import(id).then(response => {
+      const module = Object.assign({ moduleName }, response);
+      if (module.backendDependencies) {
+        checkBackendDeps(module);
+      }
+    });
+  }
+  return originalOnload.apply(this, arguments);
+};
+
+function checkBackendDeps(module: any) {
+  if (isEmpty(installedBackendModules)) {
+    fetchInstalledBackendModules().then(({ data }) => {
```

Comment:
We should handle the error here. Otherwise, we'll still have an uncaught promise exception. See https://github.com/xjamundx/eslint-plugin-promise/blob/master/docs/rules/catch-or-return.md for an explanation on why this is important.

Maybe add `.catch(err => {setTimeout(() => {throw err})})` for now. The esm-error-handling repo will surface a function in the future to better handle this.

Reply:
good catch, thanks I'll do that.